### PR TITLE
Ap 4628 - adding offline order for Magento.

### PR DESCRIPTION
**Pull request by Pro777**

### CODE REVIEW

File: app/code/community/Apruve/ApruvePayment/Model/Api/Rest/Account.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+     * Retrieve an the first instance of the buyers shopper_id
+     *
+     * @param string $mail
+     * @return $result string
+     */
+    public function getShopperId($email)
+    {
+        $corperateAccountArray = $this->_fields[0];
+        foreach ($corperateAccountArray['authorized_buyers'] as $buyer) {
+            if ($buyer['email'] === $email) {
```
Comment (CullenShane) [inline app/code/community/Apruve/ApruvePayment/Model/Api/Rest/Account.php]: emails from Apruve should be downcased, is this comparison case insensitive or downcased to match?

**Comment by CullenShane** [comment]
There's also a number of files that changed from 755 to 644 permissions.  Is there any reason for that change?

**Comment by Pro777** [comment]
@CullenShane I'll fix those permissions issue.

**Review by CullenShane — APPROVED** [review]
:work_on_my_machine:

:man_shrugging: Looks ok to me!

File: app/code/community/Apruve/ApruvePayment/Model/Api/Rest/Order.php [hunk 2]
```diff
@@ -110,30 +110,47 @@ protected function _updateOrderId($apruveOrderId, $order)
...
-        ));
-
-        $curlOptions = [];
-        $curlOptions[CURLOPT_POSTFIELDS] = $data;
-
-        $result = $this->execCurlRequest($this->_getUpdateOrderUrl($apruveOrderId), 'PUT', $curlOptions);
-        if($result['success'] == true) {
-            Mage::helper('apruvepayment')->logException('Order updated successfully...');
-            $this->_updateOrderId($apruveOrderId, $order);
+        if ($shopper_id) {
```
Comment (donnen) [inline app/code/community/Apruve/ApruvePayment/Model/Api/Rest/Order.php]: What happens when the $shopper_id is not found?

**Review by donnen — CHANGES_REQUESTED** [review]

File: app/code/community/Apruve/ApruvePayment/Model/Api/Rest/Order.php (hunk 2 above)
Comment (Pro777) [reply app/code/community/Apruve/ApruvePayment/Model/Api/Rest/Order.php]: Ok, I added code in the getShopperId method which will throw an exception if a shopper's email address is not found.

Comment (donnen) [reply app/code/community/Apruve/ApruvePayment/Model/Api/Rest/Order.php]: Okay, we just need to let the merchant know that they can't check out using Apruve if the buyer doesn't have an account.

**Comment by CullenShane** [comment]
Does the version of the plugin need to be updated?

**Pull request merged**